drivers/net/ethernet/renesas/rswitch_xenback.c

```diff
@@ -159,7 +159,7 @@ static int rswitch_vmq_back_probe(struct xenbus_device *dev,
 {
 	int err = 0;
 	struct xenbus_transaction xbt;
-	char *type_str;
+	char *type_str = NULL;
 
 	struct rswitch_vmq_back_info *be = kzalloc(sizeof(*be), GFP_KERNEL);
```

Comment:
My main concern is related to this variable, but these changes look pretty good too. I think it will be great to add one more commit, that fixes `be` variable leakage.

Reply:
Thanks. I decided to fix all leaks (there was another one, related to dormant state of a network interface) in one commit.